src/app/util/attribute-storage.cpp

```diff
@@ -1207,6 +1208,8 @@ void emAfLoadAttributeDefaults(EndpointId endpoint, bool ignoreStorage)
     uint16_t attr;
     uint8_t * ptr;
     uint16_t epCount = emberAfEndpointCount();
+    uint8_t attrData[ATTRIBUTE_LARGEST];
```

Comment:
Hopefully ATTRIBUTE_LARGEST is a reasonable size.... This may cause a very burdensome stack hit sight unseen.

Reply:
ATTRIBUTE_LARGEST is basically the larger of 8 and the longest string-typed attribute you support.

In practice seems to be 400 bytes for our apps other than all-clusters-app (which purposefully stress-tests a very large string in TestCluster, so is closer to 1KB).

I guess the question is whether to heap-allocate this instead?

Reply:
Well, heap-allocating is not necessarily much better since if it's very large, it's still very large. It boils down that this requires always to hold the largest possible string, so we have to make sure ATTRIBUTE_LARGEST is something that can have a safety added to it (i.e., to catch that an upstream change may cause your stack/heap usage to increase from N to N + M.

Reply:
Addressed in https://github.com/project-chip/connectedhomeip/pull/13600